app/views/posts/index.html.erb

```diff
@@ -1,22 +1,27 @@
 <div class="trix-content">
   <h1>Timeline</h1>
+  <%= link_to "New Post", new_post_path, class: "btn btn-primary" %>
   <% if @posts.any? %>
     <% @posts.each do |post| %>
       <div class="post">
         <h2><%= post.title %></h2>
         <p><%= post.content %></p>
-        <p> This post has <%= post.likes.count %>
-          <% if %>
-            <% post.likes.count == 1 %>
-            <p>like!<p/>
-            <% else %>
-              <p>  likes! </p>
-            <% end %>
-            <p>Posted by: <%= post.user.name %></p>
-          </div>
+        <p>This post has <%= post.likes.count %>
+          <% if post.likes.count == 1 %>
+            like!
+          <% else %>
+            likes!
+          <% end %>
+        </p>
+        <% if liked_by?(post, current_user) %>
+          <%= render "posts/partials/unlike_button", locals: {post: @post} %>
```

Comment:
I think it may not be working because you are not using either long or short form of `render`. Short:
```suggestion
          <%= render "posts/partials/unlike_button", post: @post %>
```
Long:
```suggestion
          <%= render partial: "posts/partials/unlike_button", locals: { post: @post } %>
```